support-workers/src/typescript/test/stateSchemas.test.ts

```diff
@@ -0,0 +1,85 @@
+import type {
+	DirectDebitPaymentFields,
+	StripePaymentFields,
+} from '../model/paymentFields';
+import type { CreatePaymentMethodState } from '../model/stateSchemas';
+import {
+	createPaymentMethodStateSchema,
+	createSalesforceContactStateSchema,
+} from '../model/stateSchemas';
+import createPaymentContribution from './fixtures/createPaymentMethod/contributionMonthlyUSD.json';
+import contributionWithPayPal from './fixtures/createPaymentMethod/contributionWithPayPal.json';
+import createPaymentPaper from './fixtures/createPaymentMethod/paperDirectDebit.json';
+import createPaymentSupporterPlus from './fixtures/createPaymentMethod/supporterPlusAnnualEUR.json';
+import createSalesforceContactContribution from './fixtures/createSalesforceContact/contributionMonthlyUSD.json';
+import createSalesforceContactPaper from './fixtures/createSalesforceContact/paperDirectDebit.json';
+
+describe('stateSchemas', () => {
+	test('createPaymentMethodSchema works with real inputs', () => {
+		const supporterPlus = createPaymentMethodStateSchema.parse(
+			createPaymentSupporterPlus,
+		);
+		expect(supporterPlus.product.currency).toBe('EUR');
+		expect(supporterPlus.acquisitionData?.ophanIds.pageviewId).toBe(
+			'9999999999999',
+		);
```

Comment:
What do you think about splitting each of these product/payment method tests out to their own test/it block? I think it's helpful as it makes it clearer what's going on if there are failures.